### PR TITLE
Adding tests

### PR DESCRIPTION
**Pull request by shea786**

### CODE REVIEW

File: tests/Feature/Api/RecitersApiTest.php [hunk 1]
```diff
@@ -38,7 +39,7 @@ public function testGetReciter()
     public function testGetReciterBySlug()
     {
         /** @var Reciter $reciter */
-        $reciter = factory(Reciter::class)->create();
+        $reciter = factory(\App\Reciter::class)->create();
```
Comment (szainmehdi) [inline tests/Feature/Api/RecitersApiTest.php]: Don't need this `\App\` here, because the class is imported at the top of the file.

File: database/factories/ReciterFactory.php [hunk 2]
```diff
@@ -5,14 +5,13 @@
...
     return [
         'name' => $name = $faker->name,
         'slug' => str_slug($name),
         'description' => $faker->paragraph,
         'avatar' => $faker->imageUrl(640, 480, 'people'),
         'status' => 0,
-        'created_by' => function () {
-            return factory(User::class)->create()->id;
-        },
+        'created_by' => User::all()->random()->id,
```
Comment (szainmehdi) [inline database/factories/ReciterFactory.php]: Why did you change this? This was done properly before, and now this version of it is less correct.

File: tests/Feature/Api/RecitersApiTest.php (hunk 1 above)
Comment (shea786) [reply tests/Feature/Api/RecitersApiTest.php]: i will be removing that as i was testing a few things due to some errors i was receiving

File: database/factories/ReciterFactory.php (hunk 2 above)
Comment (shea786) [reply database/factories/ReciterFactory.php]: the reason why i changed this was because currently you are creating a new user every time you run this factory, and personally i thought that was not how i thought it should be, i thought that selecting an existing user was more realistic

Comment (szainmehdi) [reply database/factories/ReciterFactory.php]: The way the `created_by` attribute was previously defined, we would create a new user if you just called the factory like this:

```php
$reciter = factory(Reciter::class)->create();
```

But if you wanted to specify which user ID, you could do that too...
```php
$user = User::find($userId);
$reciter = factory(Reciter::class)->create(['created_by' => $user->id]);
```

Now, you have it so that it will just select a random user ID. And if you haven't made sure to create a user before calling the factory, you'll see an exception.

File: database/factories/ReciterFactory.php [hunk 3]
```diff
@@ -12,7 +12,7 @@
         'avatar' => $faker->imageUrl(640, 480, 'people'),
         'status' => 0,
         'created_by' => function () {
-            return factory(User::class)->create()->id;
+        	return factory(User::class)->create()->id;
```
Comment (szainmehdi) [inline database/factories/ReciterFactory.php]: Spaces here, not tabs, but this is good

File: database/factories/AlbumFactory.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+        'reciter_id' => function () {
+        	return factory(Reciter::class)->create()->id;
+        },
+        'year' => $faker->year(),
+        'artwork' => $faker->imageUrl(640, 480, 'people'),
+        'created_by' => function () {
+        	return factory(User::class)->create()->id;
+        },
+    ];
+});
```
Comment (szainmehdi) [inline database/factories/AlbumFactory.php]: Perfect 👍

File: database/factories/TrackFactory.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+$factory->define(Track::class, function (Faker $faker) {
+    return [
+        'name' => $name = $faker->name,
+        'slug' => str_slug($name),
+        'reciter_id' => function () {
+            return factory(Reciter::class)->create();
+        },
+        'album_id' => function () {
+            return factory(Album::class)->create();
```
Comment (szainmehdi) [inline database/factories/TrackFactory.php]: return `->id()` here, like in `created_by`. Make sure all other instances of this are correct too.

File: tests/Feature/Api/TracksApiTest.php [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    public function testGetTrackById()
+    {
+        /** @var User $user */
+        /** @var Reciter $reciter */
+        /** @var Album $album */
+        /** @var Track $track */
+        $user = factory(User::class)->create();
+        $reciter = factory(Reciter::class)->create(['created_by' => $user->id]);
+        $album = factory(Album::class)->create(['created_by' => $user->id, 'reciter_id' => $reciter->id]);
+        $track = factory(Track::class)->create(['created_by' => $user->id, 'reciter_id' => $reciter->id, 'album_id' => $album->id]);
```
Comment (szainmehdi) [inline tests/Feature/Api/TracksApiTest.php:32]: Some spacing here between the creation of models and the test run would be appreciated.

**Review by szainmehdi — CHANGES_REQUESTED** [review]
See comments above.

**Review by szainmehdi — APPROVED** [review]
